### ISSUE TITLE
Popups/Calendar is not coming over the overlay

### ISSUE THREAD

**Issue by Lazeeez** [boyan01/overlay_support#86]
**Description**

If we put a calendar/Datepicker such as ShowDatePicker<https://api.flutter.dev/flutter/material/showDatePicker.html> button inside the overlay, the calendar popups behind the overlay which is not accessible. We have to close the overlay to access the calendar.

**To Reproduce**
1. Implement 'ShowOverlay' in an empty screen
2. Add ShowDatePicker inside that overlay (overlay being the parent of ShowDatePicker)
3. Try to open ShowDatePicker

**Expected behaviour**
The expected behaviour of the similar situation should be that the calendar stacks on top of the overlay and is accessible also.

**Screenshots**
<img width="1393" alt="Screenshot 2022-12-09 at 4 50 31 PM" src="https://user-images.githubusercontent.com/64376519/206691317-9cfe5781-a77b-4339-9da5-a092d0c0891b.png">

**Version (please complete the following information):**
 - Flutter Version: [e.g. v1.5.0]
 - OS: [e.g. iOS/Android]
 - OverlaySupport Version : [e.g. 0.3.0]

**Additional context**

**Comment by boyan01** [comment]
Yes, this is the expected behivior, since the global overlay should on the top of all app widgets(include dialog).

This situation seem the same as https://github.com/boyan01/overlay_support/issues/83#issuecomment-1267182999

---

According to your UI diagram, I think it is more appropriate for you to use dialog instead overlay to pop up widget in this scenario.

**Comment by Lazeeez** [comment]
I want to use overlay in my design, is there a way with which i can bound the dialog box to the overlay. i.e. the dialog box renders inside of the overlay, not behind it.

**Comment by Lazeeez** [comment]
DropDownButton also giving the same result. i programmed the DropDownButton inside the a widget and called it inside the overlay. as soon as i tap the button the menu is open behind the overlay instead of opening on top of it.

**Comment by boyan01** [comment]
Because global overlay in the widget hierarchy is more front to screen than the other app widgets (normal page, dialog, popup), so they are always show behind the overlay.  

The only way is like #83 , use  `OverlaySupport.local` to make the overlay from higher hierarchy to lower.

1. use `OverlaySupport.local` warp you current page. (Scaffold or something elese) 
2. Then call showOverlay with context parameter.

**Comment by Lazeeez** [comment]
it is still behind the overlay. Can you give me a quick implementation, maybe I'm not understanding that properly.
thanks in advance.

**Comment by Lazeeez** [comment]
I wrapped the page like you said
![Screenshot 2022-11-29 at 1 10 53 PM](https://user-images.githubusercontent.com/64376519/206706124-58f44fde-15bf-4904-a60b-403c4eddcd8e.png)
and then called the showoverlay with context.
![Screenshot 2022-12-09 at 6 20 02 PM](https://user-images.githubusercontent.com/64376519/206706288-f29e1d69-e67b-4a26-aa3a-af0db454177c.png)

is this correct?

**Comment by boyan01** [comment]
Maybe your using the wrong `context` to `showOverlay`. How about wrap the `ElevatedButton` with a `Builder` Widget, then use the `context` provided by `Builder`?

**Comment by Lazeeez** [comment]
you wanna come on a google meet? it'll be way easier and quicker that way..

**Comment by boyan01** [comment]
> you wanna come on a google meet? it'll be way easier and quicker that way..

No, I don't like talk.

> same thing.

accroding the picture, you are not using the `context` provided by `Builder`. it looks like you just changed a parameter name  of showOverlay 's builder lambda.

It should be `showOverlay((context, t), context: context1)`

**Comment by Lazeeez** [comment]
its saying overlayentry is not a widget
